src/Shared/Microsoft.DotNet.Kusto/IKustoClientProvider.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Kusto.Data.Common;
+
+namespace Microsoft.DotNet.Kusto
+{
+    public interface IKustoClientProvider
```

Comment:
Would we ever have a second implementation of this interface? What could it possibly do that would be useful?  Mocking out the entire ICslQueryProvider would be... pretty intense.

Reply:
Don't you need the interface for DI to work, or can you use a normal class?

Reply:
You can Register and Resolve any type.  Interfaces just make mocking easier.  If we _ever_ want to mock this class, we should keep the interface, since retrofitting it later would be hard, but since this thing is mostly just a "pair the options with a type", I can't imagine a scenario where that would be useful.

Reply:
I kept the interface alive for now. I think it's good to give us the flexibility if we need it later.